### PR TITLE
Added experimental/filesystem/path.cpp to libcxx (#7022)

### PR DESCRIPTION
**Pull request by Inreal**
According to issue 7022: https://github.com/kripken/emscripten/issues/7022#issuecomment-415107668

### CODE REVIEW

**Comment by dschuff** [comment]
If we're going to add path.cpp, we might as well add the rest of of std::experimental::filesystem too. Is there anything else needed (e.g. headers) to get it all?

**Comment by Inreal** [comment]
I did not try to compile the other ones, which would be quite easy but I do not have code or even tests to confirm whether they truly work. The Filesystem at least seems to work flawlessly. Do you think i should just add all of the experimental .c* files to the build script?

File: tools/system_libs.py [hunk 1]
```diff
@@ -291,7 +291,8 @@ def create_libcxx(libname):
       'utility.cpp',
       'valarray.cpp',
       'variant.cpp',
-      'vector.cpp'
+      'vector.cpp',
+      'experimental/filesystem/path.cpp'
```
Comment (kripken) [inline tools/system_libs.py]: please use `os.path.join('experimental', 'filesystem', 'path.cpp')` (that way it should work on windows too).

**Comment by kripken** [comment]
Hmm, yeah, maybe since it's marked experimental we should be careful about this, and only add it with tests (that sounds ok to me). Or it might be worth asking upstream how "experimental" that stuff is ;)

**Comment by kripken** [comment]
(By "add with tests" I mean, add a small test for this, and that sounds good to land - and other stuff later might be added also with tests.)

File: tools/system_libs.py (hunk 1 above)
Comment (Inreal) [reply tools/system_libs.py]: It works with the '/' slashes on my Windows machine, but I changed it as you proposed, looks more elegant. Thank you.

**Comment by abcthomas** [comment]
Just want to add a +1 that we should probably add the whole filesystem library. Path on its own isn't really too useful, especially without directory_iterators and the like.

**Comment by kripken** [comment]
Both suggestions sound good to me. So no need for tests, and might as well add all of experimental.

Also @Inreal please add yourself to AUTHORS. Thanks!

**Comment by kripken** [comment]
Sorry for the delay here - thanks @Inreal, that's perfect!

**Pull request merged**